rtgui/whitebalance.cc

```diff
@@ -364,6 +372,34 @@ WhiteBalance::WhiteBalance () : FoldableToolPanel(this, TOOL_NAME, M("TP_WBALANC
     equal->show ();
     tempBias->show ();
     observer10->show();
+    itcwbFrame = Gtk::manage(new Gtk::Frame(M("TP_WBALANCE_ITCWB_FRA")));
+
+    itcwbFrame->set_label_align(0.025, 0.5);
+    itcwbFrame->set_tooltip_markup (M("PREFERENCES_WBACORR_TOOLTIP"));
+
+    ToolParamBlock* const itcwbBox = Gtk::manage(new ToolParamBlock());
+
+
+    itcwb_green = Gtk::manage (new Adjuster (M("TP_WBALANCE_ITCWGREEN"), -0.35, 0.35, 0.005, 0.));
+    itcwb_green ->set_tooltip_markup (M("TP_WBALANCE_ITCWGREEN_TOOLTIP"));
+
+    itcwb_alg = Gtk::manage (new Gtk::CheckButton (M("TP_WBALANCE_ITCWB_ALG")));
+    itcwb_alg ->set_tooltip_markup (M("TP_WBALANCE_ITCWALG_TOOLTIP"));
+    itcwb_alg ->set_active (false);
+
+    itcwb_sampling = Gtk::manage (new Gtk::CheckButton (M("TP_WBALANCE_ITCWB_SAMPLING")));
+    itcwb_sampling ->set_active (false);
+
+
+    itcwb_prim = Gtk::manage (new MyComboBoxText ());
+    itcwb_prim->append(M("TP_WBALANCE_ITCWB_PRIM_SRGB"));
+    itcwb_prim->append(M("TP_WBALANCE_ITCWB_PRIM_BETA"));
+    itcwb_prim->append(M("TP_WBALANCE_ITCWB_PRIM_XYZCAM"));
+    itcwb_prim->append(M("TP_WBALANCE_ITCWB_PRIM_JDCMAX"));
+    itcwb_prim->set_active(1);
+    itcwb_primconn = itcwb_prim->signal_changed().connect(sigc::mem_fun(*this, &WhiteBalance::itcwb_prim_changed));
+    itcwb_prim ->set_tooltip_markup (M("TP_WBALANCE_ITCWPRIM_TOOLTIP"));
```

Comment:
These options were only for testing and should be removed for the release, correct?

Reply:
I keep all this data because during the development of the algorithm, it appeared that a single adjustment could not solve all the cases. by default with the option in "Preferences" disabled (white balance automatic temperature correlation disabled) 90% of cases are processed correctly, but 3 other choices are necessary to process the majority of cases, in particular "Low sampling & no use camera settings" , "Remove 2 passes algorithm" and "green refinement" (rarer)